src/tactic/noncomm_ring.lean

```diff
@@ -0,0 +1,32 @@
+/-
+Copyright (c) 2020 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import tactic.doc_commands
+import tactic.abel
+
+namespace tactic
+namespace interactive
+
+/-- A tactic for simplifying identities in not-necessarily-commutative rings. -/
+meta def noncomm_ring :=
+`[simp only [add_mul, mul_add, sub_eq_add_neg, mul_two, two_mul, pow_two, mul_assoc,
+             neg_mul_eq_neg_mul_symm, mul_neg_eq_neg_mul_symm];
+  abel]
+
+/--
+An example:
+```lean
+example {R : Type*} [ring R] (a b c : R) : a * (b + c + c - b) = 2*a*c :=
+by noncomm_ring
+```
+-/
+add_tactic_doc
```

Comment:
This is the first time I've attempted to tinker with tactics and I [do not fully understand](https://i.kym-cdn.com/entries/icons/original/000/008/342/ihave.jpg) the implications of what I pass to `add_tactic_doc` so I would be especially grateful for double checking here.

Reply:
I would move the doc string you've written here to `noncomm_ring` above. That doc string will appear as the entry for `noncomm_ring` at https://leanprover-community.github.io/mathlib_docs/tactics.html . Check that the tags you chose cohere with the existing ones there (in the drop down menu on the left). The `noncommutative` tag is unnecessary.

Reply:
Thanks, this is very helpful. I have made some adjustments based on these remarks.